rmf_site_editor/src/site/scenario.rs

```diff
@@ -18,148 +18,381 @@
+                children.iter().for_each(|e| {
+                    if let Ok((instance_modifier, affiliation)) =
+                        instance_modifiers.get(*e).map(|(i, a)| match i {
+                            InstanceModifier::Added(_) | InstanceModifier::Inherited(_) => {
+                                (InstanceModifier::new_inherited(None), a.clone())
+                            }
+                            InstanceModifier::Hidden => (i.clone(), a.clone()),
+                        })
+                    {
+                        commands
+                            .spawn(instance_modifier)
+                            .insert(affiliation)
+                            .set_parent(scenario_entity);
+                    }
+                });
+            }
+        }
+    }
 
-    if let Some(mut current_scenario) = current_scenario
-        .0
-        .and_then(|entity| scenarios.get_mut(entity).ok())
-    {
-        for (entity, pose) in changed_instances.iter() {
-            if pose.is_changed() {
-                let existing_removed_instance = current_scenario
-                    .removed_instances
-                    .iter_mut()
-                    .find(|e| **e == entity)
-                    .map(|e| e.clone());
-                if let Some(existing_removed_instance) = existing_removed_instance {
-                    current_scenario
-                        .moved_instances
-                        .retain(|(e, _)| *e != existing_removed_instance);
-                    current_scenario
-                        .added_instances
-                        .retain(|(e, _)| *e != existing_removed_instance);
-                    return;
+    // Insert instance modifier entities when new model instances are spawned and placed
+    if added_instances.is_empty() {
+        if !added_scenarios.is_empty() {
+            change_current_scenario.send(ChangeCurrentScenario(current_scenario_entity));
+        }
+        return;
+    }
+    let instance_modifier_entities =
+        get_instance_modifier_entities(current_scenario_entity, &children, &instance_modifiers);
+
+    for (instance_entity, new_pose) in added_instances.iter() {
+        if let Some((mut instance_modifier, _)) = instance_modifier_entities
+            .get(&instance_entity)
+            .and_then(|modifier_entity| instance_modifiers.get_mut(*modifier_entity).ok())
+        {
+            // If an instance modifier entity already exists for this scenario, update it
+            let instance_modifier = instance_modifier.as_mut();
+            match instance_modifier {
+                InstanceModifier::Added(_) => {
+                    *instance_modifier = InstanceModifier::new_added(new_pose.clone())
+                }
+                InstanceModifier::Inherited(_) => {
+                    *instance_modifier = InstanceModifier::new_inherited(Some(new_pose.clone()))
                 }
+                _ => {}
```

Comment:
Let's explicitly use `InstanceModifier::Hidden => {}` here just in case we ever add another variant to this enum in the future.